tests/corpus/parsing/test_cha_parser.py

```diff
@@ -0,0 +1,95 @@
+import os
+import tempfile
+import pytest
+import requests
+import sktalk
+from sktalk.corpus.parsing.cha import ChaParser
+from sktalk.corpus.parsing.parser import Parser
+
+
+class TestParser:
+    milliseconds_timestamp = [
+        ["0", "00:00:00.000"],
+        ["1706326", "00:28:26.326"],
+        ["222222", "00:03:42.222"],
+        ["None", None]
+    ]
+
+    @pytest.mark.parametrize("milliseconds, timestamp", milliseconds_timestamp)
+    def test_to_timestamp(self, milliseconds, timestamp):
+        assert Parser._to_timestamp(milliseconds) == timestamp
+
+        with pytest.raises(ValueError, match="exceeds 24h"):
+            Parser._to_timestamp("987654321")
+
+        with pytest.raises(ValueError, match="negative"):
+            Parser._to_timestamp("-1")
+
+
+class TestChaParser:
+    urls = [
+        "https://ca.talkbank.org/data-orig/GCSAusE/01.cha",
+        "https://ca.talkbank.org/data-orig/GCSAusE/02.cha",
+        "https://ca.talkbank.org/data-orig/GCSAusE/03.cha"
+    ]
+
+    @pytest.fixture(params=urls)
+    def download_file(self, request):
+        remote = request.param
+        response = requests.get(remote, timeout=5)
+        response.raise_for_status()
+
+        ext = os.path.splitext(remote)[1]
+
+        with tempfile.NamedTemporaryFile(delete=False, suffix=ext) as temp_file:
+            temp_file.write(response.content)
+            temp_file_path = temp_file.name
+
+        yield temp_file_path
+
+        os.remove(temp_file_path)
+
+    @pytest.mark.parametrize("download_file", urls, indirect=True)
+    def test_parse(self, download_file):
```

Comment:
For the sake of readability, I would try to restructure this test by adding another parameter `expected_conversation`, matching the `urls`.
That way, the test only needs to run `assert ChaParser().parse(download_file) == expected_conversation`.

This probably requires to implement the `Conversation.__eq__()` and `Conversation.__hash__()` methods, but in fact they are practically implemented within the test currently.